worksheet.rb

```diff
@@ -1,34 +1,193 @@
-
-# Use an iteration blocks to print the following answers:
-# - the number of rides each driver has given
-# - the total amount of money each driver has made
-# - the average rating for each driver
-# - Which driver made the most money?
-# - Which driver has the highest average rating?
\ No newline at end of file
+rides = [
+    {
+        driver_id: "DR0001",
+        date: [
+            "3rd Feb 2016",
+            "3rd Feb 2016",
+            "5th Feb 2016",
+        ],
+        cost: [
+            10,
+            30,
+            45,
+        ],
+        rider_id: [
+            "RD0003",
+            "RD0015",
+            "RD0003",
+        ],
+        rating: [
+            3,
+            4,
+            2,
+        ],
+    },
+    {
+        driver_id: "DR0002",
+        date: [
+            "3rd Feb 2016",
+            "4th Feb 2016",
+            "5th Feb 2016",
+        ],
+        cost: [
+            25,
+            15,
+            35,
+        ],
+        rider_id: [
+            "RD0073",
+            "RD0013",
+            "RD0066",
+        ],
+        rating: [
+            5,
+            1,
+            3,
+        ],
+    },
+
+    {
+        driver_id: "DR0003",
+        date: [
+            "4th Feb 2016",
+            "5th Feb 2016",
+        ],
+        cost: [
+            5,
+            50,
+        ],
+        rider_id: [
+            "RD0066",
+            "RD0003",
+        ],
+        rating: [
+            5,
+            2,
+        ],
+    },
+    {
+        driver_id: "DR0004",
+        date: [
+            "3rd Feb 2016",
+            "4th Feb 2016",
+            "5th Feb 2016",
+        ],
+        cost: [
+            5,
+            10,
+            20,
+        ],
+        rider_id: [
+            "RD0022",
+            "RD0022",
+            "RD0073",
+        ],
+        rating: [
+            5,
+            4,
+            5,
+        ],
+    },
+]
+
+def num_rides(ride_arr)
+  return ride_arr.map {|driver| driver[:cost].length}
+end
+
+def earn_totals(ride_arr)
+   return ride_arr.map {|driver| driver[:cost].sum}
+end
+
+def ave_rating(ride_arr)
+  return ride_arr.map do |driver|
+    driver[:rating].sum.to_f / driver[:rating].length
+  end
+end
+
+def max_earners(totals_arr)
+  new_arr = []
+  high_earning = totals_arr.max
+  totals_arr.each_with_index do |total, index|
+      if total == high_earning
+        new_arr << index
+      end
+  end
+  return new_arr
+end
+
+def max_rating(ave_arr)
+  new_arr = []
+  high_rating = ave_arr.max
+  ave_arr.each_with_index do |average, index|
+      if average == high_rating
+        new_arr << index
+      end
+  end
+  return new_arr
+end
+
+def earner_string(earnings, data)
+  string = ""
+  earnings.each_with_index do |index_pointer, index|
+    if index == 0
+      string << "#{data[index_pointer][:driver_id]}"
+    else
+      string << "&" + "#{data[index_pointer][:driver_id]}"
+  end
+    return string
+  end
+end
```

Comment:
You can simplify this using `join`:

```suggestion
def earner_string(earnings, data)
  return earnings.map { |driver| driver[:driver_id] }.join("&")
end
```

(There is also a bug here where you return after the first one, luckily there are no ties.  😃)